### PR TITLE
With chains alignment for "new" with mixins

### PR DESCRIPTION
**Pull request by poslegm**
Fixes (#1300)  and (#1495)

### CODE REVIEW

File: scalafmt-core/shared/src/main/scala/org/scalafmt/internal/Router.scala [hunk 1]
```diff
@@ -1066,8 +1053,21 @@ class Router(formatOps: FormatOps) {
...
         )
-      case tok @ FormatToken(_, right @ KwWith(), _) =>
+      case FormatToken(_, KwWith(), _) =>
         rightOwner match {
+          // something like new A with B with C
+          case template: Template if template.parent.exists { p =>
+                p.is[Term.New] || p.is[Term.NewAnonymous]
+              } =>
+            val isFirstWith = template.inits.headOption.exists { init =>
+              leftOwner.is[Init] && init == leftOwner || init.tpe == leftOwner
```
[the remark this answers is not in the document]
Comment (poslegm) [reply scalafmt-core/shared/src/main/scala/org/scalafmt/internal/Router.scala]: Exactly. Added comment with explanation

File: scalafmt-core/shared/src/main/scala/org/scalafmt/internal/FormatOps.scala [hunk 2]
```diff
@@ -736,6 +736,17 @@ class FormatOps(val tree: Tree, val initStyle: ScalafmtConfig) {
...
+  def splitWithChain(
+      isFirstWith: Boolean,
+      chain: => Set[Tree],
+      lastToken: => Token
+  ): Seq[Split] =
+    if (isFirstWith) {
+      binPackParentConstructorSplits(chain, lastToken, 2)
+    } else {
+      Seq(Split(Space, 0), Split(Newline, 1))
+    }
```
[the remark this answers is not in the document]
Comment (poslegm) [reply scalafmt-core/shared/src/main/scala/org/scalafmt/internal/FormatOps.scala]: 👌 Moved it to `Router`

File: scalafmt-core/shared/src/main/scala/org/scalafmt/internal/FormatOps.scala [hunk 3]
```diff
@@ -736,6 +736,17 @@ class FormatOps(val tree: Tree, val initStyle: ScalafmtConfig) {
     )
   }
 
+  def splitWithChain(
+      isFirstWith: Boolean,
+      chain: => Set[Tree],
+      lastToken: => Token
+  ): Seq[Split] =
+    if (isFirstWith) {
+      binPackParentConstructorSplits(chain, lastToken, 2)
```
[the remark this answers is not in the document]
Comment (poslegm) [reply scalafmt-core/shared/src/main/scala/org/scalafmt/internal/FormatOps.scala]: Done

File: scalafmt-core/shared/src/main/scala/org/scalafmt/internal/Router.scala [hunk 4]
```diff
@@ -51,6 +38,11 @@ object Constants {
   val ExceedColumnPenalty = 1000
   // Breaking a line like s"aaaaaaa${111111 + 22222}" should be last resort.
   val BreakSingleLineInterpolatedString = 10 * ExceedColumnPenalty
+  // when converting new A with B with C to
+  // new A
+  //   with B
+  //   with C
+  val IndentForWithChains = 2
```
Comment (tanishiking) [inline scalafmt-core/shared/src/main/scala/org/scalafmt/internal/Router.scala]: 👍

**Pull request merged**